src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -18,22 +19,72 @@ namespace AzureMcp.Services.Azure.Authentication;
 /// </remarks>
 public class CustomChainedCredential(string? tenantId = null) : TokenCredential
 {
-    private readonly ChainedTokenCredential _chainedCredential = CreateChainedCredential(tenantId);
+    public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        ChainedTokenCredential chainedCredential = CreateChainedCredential(tenantId);
+        return chainedCredential.GetToken(requestContext, cancellationToken);
+    }
+
+    public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        ChainedTokenCredential chainedCredential = CreateChainedCredential(tenantId);
+        return chainedCredential.GetTokenAsync(requestContext, cancellationToken);
+    }
+
+    private const string AuthenticationRecordEnvVarName = "AZURE_MCP_AUTHENTICATION_RECORD";
+    private const string OnlyUseBrokerCredentialEnvVarName = "AZURE_MCP_ONLY_USE_BROKER_CREDENTIAL";
+    private const string ClientIdEnvVarName = "AZURE_MCP_CLIENT_ID";
+
+    private static bool ShouldUseOnlyBrokerCredential()
+    {
+        string? onlyUseBrokerCredential = Environment.GetEnvironmentVariable(OnlyUseBrokerCredentialEnvVarName);
+        return onlyUseBrokerCredential == "true"
+            || onlyUseBrokerCredential == "True"
+            || onlyUseBrokerCredential == "T"
+            || onlyUseBrokerCredential == "1";
+    }
 
-    public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken) =>
-        _chainedCredential.GetToken(requestContext, cancellationToken);
+    private static ChainedTokenCredential CreateChainedCredential(string? tenantId)
```

Comment:
nit: We could have this return `TokenCredential` and name it `GetCredential` so that the implementation is agnostic.